oscap_report/scap_results_parser/scap_results_parser.py

```diff
@@ -69,5 +69,105 @@ def _debug_show_rules(self):
             logging.debug(rule_id)
             logging.debug(rule)
 
+    def _insert_rules_results(self):
+        rules_results = self.test_results.findall('.//xccdf:rule-result', NAMESPACES)
+        for rule_result in rules_results:
+            rule_id = rule_result.get('idref')
+            self.rules[rule_id].time = rule_result.get('time')
+            self.rules[rule_id].result = rule_result.find('.//xccdf:result', NAMESPACES).text
+
+            check_content_ref = rule_result.find(".//xccdf:check-content-ref", NAMESPACES)
+            if check_content_ref is not None:
+                self.rules[rule_id].oval_definition_id = check_content_ref.get("name", "")
+
+            message = rule_result.find('.//xccdf:message', NAMESPACES)
+            if message is not None:
+                self.rules[rule_id].message = message.text
+
+    @staticmethod
+    def _get_references(rule):
+        references = []
+        for referenc in rule.findall(".//xccdf:reference", NAMESPACES):
+            ref = {}
+            ref["href"] = referenc.get("href")
+            ref["text"] = referenc.text
+            references.append(ref)
+        return references
+
+    @staticmethod
+    def _get_identifiers(rule):
+        identifiers = []
+        for identifier in rule.findall(".//xccdf:ident", NAMESPACES):
+            ident = {}
+            ident["system"] = identifier.get("system")
+            ident["text"] = identifier.text
+            identifiers.append(ident)
+        return identifiers
+
+    @staticmethod
+    def _get_warnings(rule):
+        warnings = []
+        for warning in rule.findall(".//xccdf:warning", NAMESPACES):
+            warnings.append(warning.text)
+        return warnings
+
+    @staticmethod
+    def _get_remediations(rule):
+        output = []
+        for fix in rule.findall(".//xccdf:fix", NAMESPACES):
+            fix_dict = {}
+            fix_dict["remediation_id"] = fix.get("id")
+            fix_dict["system"] = fix.get("system")
+            fix_dict["complexity"] = fix.get("complexity", "")
+            fix_dict["disruption"] = fix.get("disruption", "")
+            fix_dict["strategy"] = fix.get("strategy", "")
+            fix_dict["fix"] = fix.text
+            output.append(Remediation(**fix_dict))
+        return output
+
+    @staticmethod
+    def _get_multi_check(rule):
+        for check in rule.findall(".//xccdf:check", NAMESPACES):
+            if check.get("multi-check") == "true":
```

Comment:
It would always be lower-case?

Reply:
Yes, it should always be lowercase. In [xccdf 1.2](https://csrc.nist.gov/schema/xccdf/1.2/xccdf_1.2.xsd) the `multi-check` is described in lower case and the XSLT transformation handles only `multi-check` with lower case letters.